### ISSUE TITLE
Error when using custom FabricTransportServiceRemotingProviderAttribute

### ISSUE THREAD

**Issue by antonGritsenko** [Microsoft/service-fabric-services-and-actors-dotnet#126]
Hi all

Service Fabric version: 6.0.219

We using custom FabricTransportServiceRemotingProviderAttribute with the following code:

```
public class ExtendedFabricTransportServiceRemotingProviderAttribute : FabricTransportServiceRemotingProviderAttribute
    {
        /// <summary>
        /// Create custom listener (service side) for Remoting V2 and add <see cref="ExtendedServiceRemotingDispatcher"/> to pipeline
        /// </summary>
        /// <param name="serviceContext"></param>
        /// <param name="serviceImplementation"></param>
        /// <returns></returns>
        public override IServiceRemotingListener CreateServiceRemotingListenerV2(ServiceContext serviceContext, IService serviceImplementation)
        {
            // create our own dispatcher, this will handle adding info to headers
            var messageHandler = new ExtendedServiceRemotingDispatcher(serviceContext, serviceImplementation);

            // note: attributes names remoting for V1 and V2 are NOT the same
            var listner = new FabricTransportServiceRemotingListener(serviceContext: serviceContext, serviceRemotingMessageHandler: messageHandler);

            return (IServiceRemotingListener)listner;
        }

        /// <summary>
        /// Create custom client using <see cref="ExtendedFabricTransportServiceRemotingClientFactory" />
        /// </summary>
        /// <param name="callbackMessageHandler"></param>
        /// <returns></returns>
        public override IServiceRemotingClientFactory CreateServiceRemotingClientFactoryV2(IServiceRemotingCallbackMessageHandler callbackMessageHandler)
        {
            // send default implementation as attribute
            return new ExtendedFabricTransportServiceRemotingClientFactory(base.CreateServiceRemotingClientFactoryV2(callbackMessageHandler));
        }

    }
```

This class defined in shared assembly, when used in same assembly like this:

`[assembly:ExtendedFabricTransportServiceRemotingProviderAttribute(RemotingListener = RemotingListener.V2Listener, RemotingClient = RemotingClient.V2Client)]`

This shared assembly used by 5 different services inside Service Fabric cluster.
Attribute works as expected, no problem here. 

I have issue on my dev machine (and I don't know if it same or not in production, can not check this). On each debug of the cluster (1-node config, 1 service instance per node) lines r`eturn new ExtendedFabricTransportServiceRemotingClientFactory(base.CreateServiceRemotingClientFactoryV2(callbackMessageHandler))` and `var listner = new FabricTransportServiceRemotingListener(serviceContext: serviceContext, serviceRemotingMessageHandler: messageHandler);` throws"silent"
 exception:

> System.ArgumentException
>   HResult=0x80070057
>   Message=An instance of EventSource with Guid cbd93bc2-71e5-4566-b3a7-595d8eeca6e8 already exists.
>   Source=mscorlib
>   StackTrace:
>    at System.Diagnostics.Tracing.EventSource.EnsureDescriptorsInitialized()

The service is always the same, but there is nothing special inside (I think this is just because of services order)

That's all what I have. I did check, the event source with cbd93bc2-71e5-4566-b3a7-595d8eeca6e8
 is Microsoft-ServiceFabric-Events event source (System.Fabric.Common.Tracing.FabricEvents class).

Any thoughts why it can be and how to fix this?

**Comment by suchiagicha** [comment]
@antonGritsenko 
What version of Service Fabric sdk you are using on your dev machine?
I am assuming your dev machine also has SF runtime version as  6.0.219

**Comment by antonGritsenko** [comment]
@suchiagicha sorry, forgot to mention. SDK version is 2.8.219.9494. And yes, runtime is 6.0.219.9494
BTW, We plan to upgrade it to latest at next week.

**Comment by suchiagicha** [comment]
@antonGritsenko 
I couldn't repro this issue. 
How did you found this exception ?
What is your service replica status using below command , what does it show?

PS C:\windows\system32> Get-ServiceFabricApplication|Get-ServiceFabricService|Get-ServiceFabricPartition|Get-ServiceFabricReplica

Could you share your sample with us?

**Comment by suchiagicha** [comment]
@antonGritsenko 
Let me rephrase the question. Is it a firstchance exception /(handled exception) or unhandled exception?

**Comment by antonGritsenko** [comment]
This is unhandled exception. 
BTW, I think it's only occurs when two or more services used same shared library. I still have no time to create repro app, plan to do it later this week.

**Comment by antonGritsenko** [comment]
here we go: https://github.com/antonGritsenko/SF.CustomRemoting
Please enable System.ArgumentException in Exception settings

**Comment by suchiagicha** [comment]
@antonGritsenko 
Why you need to enable any Exception in ExceptionSettings?
What happens if you dont?
If you are not seeing exception without enabling it  , it means its a handled exception.

**Comment by suchiagicha** [comment]
Closing this issue as there is no communication .
